packages/dialect-react-ui/components/CardinalAddress/index.tsx

```diff
@@ -0,0 +1,97 @@
+import { useAddressName } from '@cardinal/namespaces-components';
+import { TwitterIcon } from '../Icon/Twitter';
+import type { Connection } from '@project-serum/anchor';
+import type { PublicKey } from '@solana/web3.js';
+
+const formatTwitterLink = (handle: string | undefined, color: string) => {
+  if (!handle) return <a></a>;
+  return (
+    <a
+      href={`https://twitter.com/${handle}`}
+      style={{ color: color }}
+      target="_blank"
+      rel="noreferrer"
+    >
+      {handle}
+    </a>
+  );
+};
+
+function shortenAddress(address: string, chars = 5): string {
+  return `${address.substring(0, chars)}...${address.substring(
+    address.length - chars
+  )}`;
+}
+
+const formatShortAddress = (address: PublicKey | undefined) => {
+  if (!address) return <></>;
+  return (
+    <a
+      href={`https://explorer.solana.com/address/${address.toString()}`}
+      target="_blank"
+      rel="noopener noreferrer"
+    >
+      {shortenAddress(address.toString())}
+    </a>
+  );
+};
+
+const DisplayAddressNew = ({
+  connection,
+  address,
+  height = "13",
+  width = "300",
+  style,
+  color = "#FFFFFF",
+}: {
+  connection: Connection;
+  address: PublicKey | undefined;
+  height?: string;
+  width?: string;
+  style?: React.CSSProperties;
+  color?: string
+}) => {
+  const { displayName, loadingName } = useAddressName(connection, address);
+
+  if (!address) return <></>;
+  return loadingName ? (
+    <div
+      style={{
+        ...style,
+        height,
+        width,
+        overflow: "hidden",
```

Comment:
minor: let's run the prettier against new files using the .prettierrc from the root

Reply:
ran prettier on the dialect-react-ui/*